test/extended/util/prometheus/helpers.go

```diff
@@ -134,22 +135,38 @@ func StripLabels(m model.Metric, names ...string) model.LabelSet {
 }
 
 func RunQuery(query, ns, execPodName, baseURL, bearerToken string) (*PrometheusResponse, error) {
-	url := fmt.Sprintf("%s/api/v1/query?%s", baseURL, (url.Values{"query": []string{query}}).Encode())
-	contents, err := GetBearerTokenURLViaPod(ns, execPodName, url, bearerToken)
+	queryUrl := fmt.Sprintf("%s/api/v1/query?%s", baseURL, (url.Values{"query": []string{query}}).Encode())
+	return runQuery(queryUrl, ns, execPodName, bearerToken)
+}
+
+func RunQueryAtTime(query, ns, execPodName, baseURL, bearerToken string, evaluationTime model.Time) (*PrometheusResponse, error) {
+	queryParams := url.Values{"query": []string{query}, "time": []string{evaluationTime.String()}}
+	queryUrl := fmt.Sprintf("%s/api/v1/query?%s", baseURL, queryParams.Encode())
+	result, err := runQuery(queryUrl, ns, execPodName, bearerToken)
+	if err != nil {
+		return nil, fmt.Errorf("error executing query %s: %v", query, err)
+	}
+
+	return result, nil
+}
+
+func runQuery(queryUrl, ns, execPodName, bearerToken string) (*PrometheusResponse, error) {
+	contents, err := GetBearerTokenURLViaPod(ns, execPodName, queryUrl, bearerToken)
 	if err != nil {
-		return nil, fmt.Errorf("unable to execute query %s: %v", query, err)
+		return nil, fmt.Errorf("unable to execute query %v", err)
 	}
 
 	// check query result, if this is a new error log it, otherwise remain silent
 	var result PrometheusResponse
 	if err := json.Unmarshal([]byte(contents), &result); err != nil {
-		return nil, fmt.Errorf("unable to parse query response for %s: %v", query, err)
+		return nil, fmt.Errorf("unable to parse query response: %v", err)
 	}
 	metrics := result.Data.Result
 	if result.Status != "success" {
 		data, _ := json.MarshalIndent(metrics, "", "  ")
-		return nil, fmt.Errorf("promQL query: %s had reported incorrect status:\n%s", query, data)
+		return nil, fmt.Errorf("incorrect response status: %s with error %s", data, result.Error)
```

Comment:
Would it be good to log which query was reporting incorrect response status?

Reply:
Good point. 

In this function we have the full prometheus URL so I decided to wrap the error in the calling functions [here](https://github.com/openshift/origin/pull/26162/files#diff-e46d7d1c9040bf6b47bd6b4f9c10e2b511f0ec30b66e763523503ee9523c613dR139) and [here](https://github.com/openshift/origin/pull/26162/files#diff-e46d7d1c9040bf6b47bd6b4f9c10e2b511f0ec30b66e763523503ee9523c613dR150) to make sure we don't pollute the logs and improve readability of errors.